test/jasmine/tests/violin_test.js

```diff
@@ -724,6 +724,40 @@ describe('Test violin hover:', function() {
             .then(done);
         });
     });
+
+    it('labels should avoid overlaps', function(done) {
+        gd = createGraphDiv();
+
+        var fig = Lib.extendDeep({}, require('@mocks/violin_zoomed-in.json'));
+        fig.layout.width = 700;
+        fig.layout.height = 450;
+
+        Plotly.plot(gd, fig)
```

Comment:
I couldn't reproduce the bug using a _simpler_ mock, so I added the data/layout used in https://codepen.io/etpinard/pen/wberrP?editors=1010 as a mock. 

This test here currently yields

![image](https://user-images.githubusercontent.com/6675409/58206563-e59df980-7cae-11e9-945e-210f51e40166.png)


on master.